src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
> But, in Java, one never changes the behaviour of equals without changing the hashCode consistently

That's mainly because most collection types rely on that contract. But we don't actually change the implementation of equals, we only change the way java-object-diff determines the object identity.